src/lib.rs

```diff
@@ -71,33 +84,33 @@ pub struct ImageSize {
 ///
 /// This will check the header to determine what image type the data is.
 pub fn image_type(header: &[u8]) -> ImageResult<ImageType> {
-    if header.len() >= 2 {
-        if &header[0..2] == b"\x42\x4D" {
-            return Ok(ImageType::Bmp);
-        } else if header.len() >= 3 && &header[0..3] == b"\xFF\xD8\xFF" {
-            return Ok(ImageType::Jpeg);
-        } else if header.len() >= 4 && &header[0..4] == b"\x89PNG" {
-            return Ok(ImageType::Png);
-        } else if header.len() >= 4 && &header[0..4] == b"GIF8" {
-            return Ok(ImageType::Gif);
-        } else if header.len() >= 4 && (&header[0..4] == b"II\x2A\x00" || &header[0..4] == b"MM\x00\x2A") {
-            return Ok(ImageType::Tiff);
-        } else if header.len() >= 4 && &header[0..4] == b"8BPS" {
-            return Ok(ImageType::Psd);
-        } else if header.len() >= 8 && &header[4..8] == b"ftyp" {
-            return Ok(ImageType::Heif);
-        } else if header.len() >= 12 && &header[0..4] == b"RIFF" && &header[8..12] == b"WEBP" {
-            return Ok(ImageType::Webp);
-        } else if (header.len() >= 2 && &header[0..2] == b"\xFF\x0A")
-            || (header.len() >= 12 && &header[0..12] == b"\x00\x00\x00\x0CJXL \x0D\x0A\x87\x0A")
-        {
-            return Ok(ImageType::Jxl);
-        } else {
-            return Err(ImageError::NotSupported);
-        }
+    if header.len() < 2 {
+        Err(std::io::Error::new(std::io::ErrorKind::UnexpectedEof, "Not enough data").into())
+    } else if header.starts_with(b"\x42\x4D") {
```

Comment:
I changed these to use [`starts_with()`](https://doc.rust-lang.org/std/primitive.slice.html#method.starts_with) which is a bit easier to read IMO. If you want I can revert it and make the minimal changes for ICO support.

Reply:
This is fine, although I do see one [0..4] for RIFF still there. Not sure if this was just skipped over or kept on purpose.

Reply:
kept on purpose, just because it's right next to another slice comparison

Reply:
kept on purpose, just because it's right next to another slice comparison